src/plugins/unified_field_list/public/components/field_item_button/field_item_button.tsx

```diff
@@ -22,72 +22,211 @@ import './field_item_button.scss';
 }
 
 /**
- * Inner part of field list item
+ * Field list item component
  * @param field
  * @param fieldSearchHighlight
+ * @param isSelected
  * @param isActive
  * @param isEmpty
  * @param infoIcon
  * @param className
  * @param getCustomFieldType
+ * @param dataTestSubj
+ * @param size
  * @param onClick
+ * @param shouldAlwaysShowAction
+ * @param buttonAddFieldToWorkspaceProps
+ * @param buttonRemoveFieldFromWorkspaceProps
+ * @param onAddFieldToWorkspace
+ * @param onRemoveFieldFromWorkspace
  * @param otherProps
  * @constructor
  */
 export function FieldItemButton<T extends FieldListItem = DataViewField>({
   field,
   fieldSearchHighlight,
+  isSelected,
   isActive,
   isEmpty,
   infoIcon,
   className,
   getCustomFieldType,
+  dataTestSubj,
+  size,
   onClick,
+  shouldAlwaysShowAction,
+  buttonAddFieldToWorkspaceProps,
+  buttonRemoveFieldFromWorkspaceProps,
+  onAddFieldToWorkspace,
+  onRemoveFieldFromWorkspace,
   ...otherProps
 }: FieldItemButtonProps<T>) {
   const displayName = field.displayName || field.name;
+  const title =
+    displayName !== field.name && field.name !== '___records___'
+      ? i18n.translate('unifiedFieldList.fieldItemButton.fieldTitle', {
+          defaultMessage: '{fieldDisplayName} ({fieldName})',
+          values: {
+            fieldName: field.name,
+            fieldDisplayName: displayName,
+          },
+        })
+      : displayName;
+
   const iconProps = getCustomFieldType
     ? { type: getCustomFieldType(field) }
     : getFieldIconProps(field);
   const type = iconProps.type;
 
   const classes = classnames(
-    'unifiedFieldItemButton',
+    'unifiedFieldListItemButton',
     {
-      [`unifiedFieldItemButton--${type}`]: type,
-      [`unifiedFieldItemButton--exists`]: !isEmpty,
-      [`unifiedFieldItemButton--missing`]: isEmpty,
+      [`unifiedFieldListItemButton--${type}`]: type,
+      [`unifiedFieldListItemButton--exists`]: !isEmpty,
+      [`unifiedFieldListItemButton--missing`]: isEmpty,
     },
     className
   );
 
+  const addFieldToWorkspaceTooltip =
+    buttonAddFieldToWorkspaceProps?.['aria-label'] ??
+    i18n.translate('unifiedFieldList.fieldItemButton.addFieldToWorkspaceLabel', {
+      defaultMessage: 'Add "{field}" field',
+      values: {
+        field: field.displayName,
+      },
+    });
+
+  const removeFieldFromWorkspaceTooltip =
+    buttonRemoveFieldFromWorkspaceProps?.['aria-label'] ??
+    i18n.translate('unifiedFieldList.fieldItemButton.removeFieldToWorkspaceLabel', {
+      defaultMessage: 'Remove "{field}" field',
+      values: {
+        field: field.displayName,
+      },
+    });
+
+  const fieldActionClassName = classnames('unifiedFieldListItemButton__action', {
+    'unifiedFieldListItemButton__action--always': shouldAlwaysShowAction,
+  });
+  const fieldAction = isSelected
+    ? onRemoveFieldFromWorkspace && (
+        <EuiToolTip
+          key={`selected-to-remove-${field.name}-${removeFieldFromWorkspaceTooltip}`}
```

Comment:
Question, nothing blocking, is such a long `key` necessary? Is there even a key necessary?

Reply:
@kertal Yes, the tooltip component needs to be recreated otherwise it was appearing incorrectly after pressing a button.